lib/taiko.js

```diff
@@ -1909,30 +1909,55 @@ function match(text, ...args) {
         let textToTranslate = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ' + nbspChar;
         let translateTo = 'abcdefghijklmnopqrstuvwxyz' + ' ';
         let xpathText = `translate(normalize-space(${xpath(text)}), "${textToTranslate}", "${translateTo}")`;
-        if (e === '*') elements = await matchTextNode(textToTranslate, translateTo, xpathText);
-        if (!elements || !elements.length) elements = await matchValueOrType(e, textToTranslate, translateTo, xpathText);
-        if (!elements || !elements.length) elements = await matchTextAcrossElements(e, textToTranslate, translateTo, xpathText);
+        elements = await matchExactElements(e, textToTranslate, translateTo, xpathText);
+        if (!elements || !elements.length) elements = await matchContainsElements(e, textToTranslate, translateTo, xpathText);
         return await handleRelativeSearch(elements, args);
     };
     return { get: getIfExists(get), exists: exists(getIfExists(get), descEvent), description: `Element matching text "${text}"`, text: selectorText(get) };
 }
 
-async function matchTextNode(textToTranslate, translateTo, xpathText) {
-    const elements = await $$xpath(`//text()[translate(normalize-space(string()), "${textToTranslate}", "${translateTo}")=${xpathText}]`);
-    if ((!elements || !elements.length)) return await $$xpath(`//text()[contains(translate(normalize-space(string()), "${textToTranslate}", "${translateTo}")=${xpathText})]`);
+async function matchExactElements(e, textToTranslate, translateTo, xpathText) {
```

Comment:
can these methods be moved into a separate file as they are well isolated now ? something like textSearch.js